lib/flake.nix

```
@@ -0,0 +1,90 @@
+{
+  description = "DevOS environment configuriguration library.";
+
+  inputs =
+    {
+      deploy.url = "github:serokell/deploy-rs";
+      deploy.inputs = {
+        utils.follows = "utils";
+      };
+      devshell.url = "github:numtide/devshell";
+      utils.url = "github:numtide/flake-utils";
+      
+    };
+
+  outputs = inputs@{ self, nixpkgs, deploy, devshell, utils, ... }: let
+
+    lib = nixpkgs.lib.makeExtensible (self:
+      let
+        attrs = import ./attrs.nix { lib = nixpkgs.lib // self; };
+        lists = import ./lists.nix { lib = nixpkgs.lib // self; };
+        strings = import ./strings.nix { lib = nixpkgs.lib // self; };
+      in
+
+      utils.lib
+
+      //
+
+      {
+        os = import ./devos {
+          lib = nixpkgs.lib // self;
+          inherit utils;
+        };
+
+        mkFlake = {
+          __functor = import ./mkFlake {
+            lib = nixpkgs.lib // self;
+            inherit deploy;
+          };
+          evalArgs = import  ./mkFlake/evalArgs.nix { lib = nixpkgs.lib // self; };
+          evalOldArgs = import  ./mkFlake/evalOldArgs.nix { lib = nixpkgs.lib // self; };
+        };
+
+        pkgs-lib = import ./pkgs-lib {
+          lib = nixpkgs.lib // self;
+          inherit nixpkgs deploy devshell;
+        };
+
+        inherit (attrs)
+          mapFilterAttrs
+          genAttrs'
+          safeReadDir
+          pathsToImportedAttrs
+          concatAttrs
+          filterPackages;
+        inherit (lists) pathsIn;
+        inherit (strings) rgxToString;
+      }
+    );
+
+    jobs = import ./jobs { inherit nixpkgs; lib = nixpkgs.lib // lib; };
+
+  in
+
+  {
+    inherit jobs;
+
+    lib = utils.lib // {
+      inherit (lib)
+        mkFlake;
+    };
```

```suggestion
  inherit lib;
```
Is there a reason not to export the entire devos lib right now? Or even just move the lib down here and refer to it as `self.lib` when passing to checks.

---

Nope, not really. I think before we finalize we should go over this again and only export stuff for which we identify a clear use case.

This, in oder to only commit (resources to maintain) a public api where we create actual value.